### PR TITLE
improve error reporting and add a test for that

### PR DESCRIPTION
**Pull request by cosnicolaou**
this PR makes sure that meaningful errors messages for ignored functions as well as methods, and also that the error message contains the name of the function and/or receiver. A test is also included for these errors.

### CODE REVIEW

File: _examples/gopyerrors/test.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2020 go-python Authors.  All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+# py2/py3 compat
+from __future__ import print_function
+
+import gopyerrors
+
+# this is empty, it's only purpose is to have
```
Comment (sbinet) [inline _examples/gopyerrors/test.py]:
```suggestion
# this is empty, its sole purpose is to have
```

File: main_test.go [hunk 2]
```diff
@@ -90,6 +90,37 @@ func TestGofmt(t *testing.T) {
...
 }
 
+func TestGoPyErrors(t *testing.T) {
+	pyvm := testBackends["py3"]
+	workdir, err := ioutil.TempDir("", "gopy-")
+	if err != nil {
+		t.Fatalf("could not create workdir: %v\n", err)
+	}
+	fmt.Printf("pyvm: %s making work dir: %s\n", pyvm, workdir)
+	//defer os.RemoveAll(workdir)
```
Comment (sbinet) [inline main_test.go]: uncomment?

File: main_test.go [hunk 3]
```diff
@@ -90,6 +90,37 @@ func TestGofmt(t *testing.T) {
...
+
+	curPkgPath := reflect.TypeOf(pkg{}).PkgPath()
+
+	// fmt.Printf("building in work dir: %s\n", workdir)
+	fpath := filepath.Join(curPkgPath, "_examples/gopyerrors")
+	args := []string{"run", ".", "gen", "-vm=" + pyvm, "-output=" + workdir, fpath}
+	cmd := exec.Command("go", args...)
+	fmt.Printf("running: %v\n", args)
```
Comment (sbinet) [inline main_test.go]:
```suggestion
	t.Logf("running: %v\n", cmd.Args)
```
(and perhaps remove the `args` var alltogether?)

File: main_test.go [hunk 4]
```diff
@@ -90,6 +90,37 @@ func TestGofmt(t *testing.T) {
...
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("cmd: go %v %v\n", strings.Join(args, " "), err)
```
Comment (sbinet) [inline main_test.go]:
```suggestion
		t.Fatalf("could not run %v: %+v\n", strings.Join(cmd.Args, " "), err)
```

File: main_test.go [hunk 5]
```diff
@@ -90,6 +90,37 @@ func TestGofmt(t *testing.T) {
...
+	}
+	contains := `--- Processing package: github.com/go-python/gopy/_examples/gopyerrors ---
+ignoring python incompatible function: gopyerrors.func github.com/go-python/gopy/_examples/gopyerrors.NotErrorMany() (int, int): func() (int, int): gopy: second result value must be of type error: func() (int, int)
+ignoring python incompatible method: gopyerrors.func (*github.com/go-python/gopy/_examples/gopyerrors.Struct).NotErrorMany() (int, string): func() (int, string): gopy: second result value must be of type error: func() (int, string)
+ignoring python incompatible method: gopyerrors.func (*github.com/go-python/gopy/_examples/gopyerrors.Struct).TooMany() (int, int, string): func() (int, int, string): gopy: too many results to return: func() (int, int, string)
+ignoring python incompatible function: gopyerrors.func github.com/go-python/gopy/_examples/gopyerrors.TooMany() (int, int, string): func() (int, int, string): gopy: too many results to return: func() (int, int, string)
+`
+	if got, want := string(out), contains; !strings.Contains(got, want) {
+		t.Errorf("%v does not contain %v\n", got, want)
```
Comment (sbinet) [inline main_test.go]:
```suggestion
		t.Fatalf("%v does not contain %v\n", got, want)
```

File: main_test.go [hunk 6]
```diff
@@ -90,6 +90,37 @@ func TestGofmt(t *testing.T) {
 	}
 }
 
+func TestGoPyErrors(t *testing.T) {
+	pyvm := testBackends["py3"]
+	workdir, err := ioutil.TempDir("", "gopy-")
+	if err != nil {
+		t.Fatalf("could not create workdir: %v\n", err)
+	}
+	fmt.Printf("pyvm: %s making work dir: %s\n", pyvm, workdir)
```
Comment (sbinet) [inline main_test.go]:
```suggestion
	t.Logf("pyvm: %s making work dir: %s\n", pyvm, workdir)
```

File: main_test.go [hunk 7]
```diff
@@ -90,6 +90,37 @@ func TestGofmt(t *testing.T) {
...
+	//defer os.RemoveAll(workdir)
+
+	curPkgPath := reflect.TypeOf(pkg{}).PkgPath()
+
+	// fmt.Printf("building in work dir: %s\n", workdir)
```
Comment (sbinet) [inline main_test.go]:
```suggestion
```

File: main_test.go [hunk 8]
```diff
@@ -90,6 +90,37 @@ func TestGofmt(t *testing.T) {
...
+	fpath := filepath.Join(curPkgPath, "_examples/gopyerrors")
+	args := []string{"run", ".", "gen", "-vm=" + pyvm, "-output=" + workdir, fpath}
+	cmd := exec.Command("go", args...)
```
Comment (sbinet) [inline main_test.go]:
```suggestion
	cmd := exec.Command("go", "run", ".", "gen", "-vm=" + pyvm, "-output=" + workdir, fpath)
```

File: main_test.go [hunk 9]
```diff
@@ -90,6 +90,37 @@ func TestGofmt(t *testing.T) {
...
+		t.Fatalf("could not create workdir: %v\n", err)
+	}
+	fmt.Printf("pyvm: %s making work dir: %s\n", pyvm, workdir)
+	//defer os.RemoveAll(workdir)
+
+	curPkgPath := reflect.TypeOf(pkg{}).PkgPath()
+
+	// fmt.Printf("building in work dir: %s\n", workdir)
+	fpath := filepath.Join(curPkgPath, "_examples/gopyerrors")
+	args := []string{"run", ".", "gen", "-vm=" + pyvm, "-output=" + workdir, fpath}
```
Comment (sbinet) [inline main_test.go]:
```suggestion
```

File: main_test.go [hunk 10]
```diff
@@ -96,19 +96,16 @@ func TestGoPyErrors(t *testing.T) {
...
 	fpath := filepath.Join(curPkgPath, "_examples/gopyerrors")
-	args := []string{"run", ".", "gen", "-vm=" + pyvm, "-output=" + workdir, fpath}
-	cmd := exec.Command("go", args...)
-	fmt.Printf("running: %v\n", args)
+	cmd := exec.Command("go", "run", ".", "gen", "-vm="+pyvm, "-output="+workdir, fpath)
+	t.Logf("running: %v\n", cmd.Args)
 	out, err := cmd.CombinedOutput()
 	if err != nil {
-		t.Fatalf("cmd: go %v %v\n", strings.Join(args, " "), err)
+		t.Fatalf("could not run %v %v\n", strings.Join(cmd.Args, " "), err)
```
Comment (sbinet) [inline main_test.go]:
```suggestion
		t.Fatalf("could not run %v: %+v\n", strings.Join(cmd.Args, " "), err)
```
(to leverage the new error-wrapping of Go-1.13. one day)

Comment (cosnicolaou) [reply main_test.go]: done!

**Pull request merged**